packages/core/src/components/tree/treeNode.tsx

```diff
@@ -13,7 +13,7 @@ import { safeInvoke } from "../../common/utils";
 import { Collapse } from "../collapse/collapse";
 import { Icon, IconName } from "../icon/icon";
 
-export interface ITreeNode extends IProps {
+export interface ITreeNode<T = {}> extends IProps {
```

Comment:
can we say `T = undefined`? more accurate for the default case. if you want to use `<T>` then you have to define it.

Reply:
When changing the generic constrain to <T = undefined>,  each call to safeInvoke() won't complied:
Argument of type 'this' is not assignable to parameter of type 'TreeNode<undefined>'.

I don't see the problem with the <T = {}> constrain, the user can still use the tree without declaring on any type.

Reply:
This was my original intent too, but per @adidahiya requested 16 days ago: 
"you can set a default for the generic type parameter so that users who don't need it won't have to specify T. 
use nodeData: T where the interface is defined as ITreeNode<T = {}>, similar to queryList:
blueprint/packages/select/src/components/query-list/queryList.tsx (Line 15 in ff76c4a)"

since both of you are reviewers and contributors, I suggest you guys touch bases and sync your request.

Reply:
i guess `{}` is fine, since React does the same thing and `state` doesn't actually default to anything (you gotta initialize it yourself)